processor/spanmetricsprocessor/processor.go

```diff
@@ -249,39 +274,56 @@ func (p *processorImp) ConsumeTraces(ctx context.Context, traces pdata.Traces) e
 // writes the raw metrics data into the metrics object.
 func (p *processorImp) buildMetrics() (*pdata.Metrics, error) {
 	m := pdata.NewMetrics()
-	ilm := m.ResourceMetrics().AppendEmpty().InstrumentationLibraryMetrics().AppendEmpty()
-	ilm.InstrumentationLibrary().SetName("spanmetricsprocessor")
+	rms := m.ResourceMetrics()
+	for _, key := range p.resourceKeyToDimensions.Keys() {
+		cachedResourceAttributesMap, ok := p.resourceKeyToDimensions.Get(key)
+		if !ok {
+			return nil, errors.New("expected cached resource attributes not found")
+		}
 
-	// Obtain write lock to reset data
-	p.lock.Lock()
+		resourceAttributesMap, ok := cachedResourceAttributesMap.(pdata.AttributeMap)
+		if !ok {
+			return nil, errors.New("expected cached resource attributes type assertion failed")
+		}
 
-	if err := p.collectCallMetrics(ilm); err != nil {
-		p.lock.Unlock()
-		return nil, err
-	}
+		// If the service name doesn't exist, we treat it as invalid and do not generate a metric
```

Comment:
We are changing this field to be mandatory. See the PR description:

> Service Name has been moved to become a default resource attribute instead of the attribute as per the resource semantic convention.